### PR TITLE
Add CI support for code coverage.

### PR DESCRIPTION
**Pull request by oSoMoN**

### CODE REVIEW

**Comment by oSoMoN** [comment]
There's some missing setup required by codecov.io to integrate with github CI, which requires action from an owner of the repository. I'm guessing this is the reason why the CI check currently doesn't show up here?

File: .github/workflows/flutter-ci.yml [hunk 1]
```diff
@@ -38,7 +38,13 @@ jobs:
     
     - name: Run tests on subiquity_client
       working-directory: ./packages/subiquity_client
-      run: dart test
+      run: flutter test --coverage
```
Comment (MarcusTomlinson) [inline .github/workflows/flutter-ci.yml:41]: This looks like a mistake. Should it not be `dart test --coverage`?

**Review by MarcusTomlinson — CHANGES_REQUESTED** [review]

**Comment by MarcusTomlinson** [comment]
I assume uploading multiple sets of coverage in CI is common practice? I don’t know enough about this. Does the report get generated as a combination of the two coverage results?

**Comment by oSoMoN** [comment]
> I assume uploading multiple sets of coverage in CI is common practice? I don’t know enough about this. Does the report get generated as a combination of the two coverage results?

I lack experience with this, but [the documentation](https://docs.codecov.io/docs/merging-reports) seems to imply that reports will be merged automatically. Let's see if it happens in practice…

**Comment by MarcusTomlinson** [comment]
Ok looks alright then. Let’s see how it goes.

**Pull request merged**